templates/types/streaming/fastapi/app/api/routers/vercel_response.py

```diff
@@ -47,25 +41,24 @@ async def content_generator(
         cls,
         request: Request,
         event_handler: EventCallbackHandler,
-        response: StreamingAgentChatResponse,
+        response: Awaitable[StreamingAgentChatResponse],
         chat_data: ChatData,
+        background_tasks: BackgroundTasks,
     ):
+        # Yield the events from the event handler
+        async def _event_generator():
```

Comment:
this code was never easy to read :D 
how about moving out the two generator functions?